cli/src/commands/branch/create.rs

```diff
@@ -42,14 +43,22 @@ pub fn cmd_branch_create(
         workspace_command.resolve_single_rev(args.revision.as_ref().unwrap_or(&RevisionArg::AT))?;
     let view = workspace_command.repo().view();
     let branch_names = &args.names;
-    if let Some(branch_name) = branch_names
-        .iter()
-        .find(|&name| view.get_local_branch(name).is_present())
-    {
-        return Err(user_error_with_hint(
-            format!("Branch already exists: {branch_name}"),
-            "Use `jj branch set` to update it.",
-        ));
+    for name in branch_names {
+        if view.get_local_branch(name).is_present() {
+            return Err(user_error_with_hint(
+                format!("Branch already exists: {name}"),
+                "Use `jj branch set` to update it.",
+            ));
+        }
+        if has_tracked_remote_branches(view, name) {
+            return Err(user_error_with_hint(
+                format!("Tracked remote branches exist for deleted branch: {name}"),
+                format!(
+                    "Use `jj branch set` to reset local branch. Run `jj branch untrack \
```

Comment:
"reset" is a bit confusing here, let's do "recreate"?

Reply:
It's not "creating" new branch (in a sense of `jj branch create`), but seems okay here. Updated.